app/src/modules/visual/routes/visual-editor.vue

```diff
@@ -0,0 +1,104 @@
+<script setup lang="ts">
+import { ref, computed } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useHead } from '@unhead/vue';
+import { useSettingsStore } from '@/stores/settings';
+import ModuleBar from '@/views/private/components/module-bar.vue';
+import NotificationDialogs from '@/views/private/components/notification-dialogs.vue';
+import NotificationsGroup from '@/views/private/components/notifications-group.vue';
+import LivePreview from '@/views/private/components/live-preview.vue';
+import EditingLayer from '../components/editing-layer.vue';
+
+const { t } = useI18n();
+useHead({ title: t('visual_editor') });
+
+const { settings } = useSettingsStore();
+
+const urls = computed<string[]>(() => {
+	const urls: string[] = [];
+
+	if (settings?.visual_editor_project_url && settings?.project_url) {
+		urls.push(settings.project_url);
+	}
+
+	settings?.visual_editor_urls?.forEach((item) => {
+		if (item.url) urls.push(item.url);
+	});
+
+	return urls;
+});
+
+const moduleBarOpen = ref(true);
+const showEditableElements = ref(false);
+</script>
+
+<template>
+	<div class="module">
+		<transition-expand x-axis>
+			<module-bar v-if="moduleBarOpen" />
+		</transition-expand>
+
+		<live-preview :url="urls" :header-expanded="moduleBarOpen" hide-popup-button>
+			<template #prepend-header>
```

Comment:
If the intention is to have `visual_editor_urls` be the "sites" you can go and do visual editing on, then there needs to be navigation buttons and possibly the ability to type in the path you want to go to. There currently is no way to navigate back/forward or go to specific page on a website without interacting with your website, which may not always have a way to get to specific pages.

If the intention is to configre every URL/Page you would need to edit using the visual editor into settings then this works as expected, however, a giant dropdown list of URL's does not seem very UX friendly.